packages/measured-signalfx-reporter/lib/reporters/SignalFxMetricsReporter.js

```diff
@@ -0,0 +1,222 @@
+const { Reporter } = require('measured-reporting');
+const { MetricTypes } = require('measured-core');
+const { validateSignalFxClient } = require('../validators/inputValidators');
+
+/**
+ * A Reporter that reports metrics to Signal Fx
+ * @extends {Reporter}
+ */
+class SignalFxMetricsReporter extends Reporter {
+  /**
+   * @param {SignalFxClient} signalFxClient The configured signal fx client.
+   * @param {ReporterOptions} [options] See {@link ReporterOptions}.
+   */
+  constructor(signalFxClient, options) {
+    super(options);
+    validateSignalFxClient(signalFxClient);
+    this._signalFxClient = signalFxClient;
+  }
+
+  /**
+   * Sends metrics to signal fx, converting name and dimensions and {@link Metric} to data signal fx can ingest
+   * @param {MetricWrapper[]} metrics The array of metrics to send to signal fx.
+   * @protected
+   */
+  _reportMetrics(metrics) {
+    this._log.debug('_reportMetrics() called');
+
+    let signalFxDataPointRequest = {};
+
+    metrics.forEach(metric => {
+      if (!metric) {
+        this._log.warn('Metric was null when it should not have been');
+        return;
+      }
+      signalFxDataPointRequest = this._processMetric(metric, signalFxDataPointRequest);
+    });
+
+    this._log.debug('Sending data to Signal Fx');
+
+    this._signalFxClient.send(signalFxDataPointRequest).catch(error => {
+      this._log.error('Failed to send metrics to signal fx error:', error);
+    });
+  }
+
+  /**
+   * Method for getting raw signal fx api request values from the Timer Object.
+   *
+   * @param {MetricWrapper} metric metric The Wrapped Metric Object.
+   * @param {any} currentBuiltRequest The signal fx request that is being built.
+   * @return {any} the currentBuiltRequest The signal fx request that is being built with the given metric in it.
```

Comment:
nitpick: For consistency, I would convert to "SignalFx" everywhere, matches how they have their name on their website.